lib/spyfall_web/live/game_live.ex

```diff
@@ -32,6 +35,11 @@ defmodule SpyfallWeb.GameLive do
     {:noreply, socket}
   end
 
+  def handle_event("new_name", %{"value" => new_name}, socket) do
+    Player.update(self(), "game:" <> socket.assigns.game, socket.id, %{name: new_name})
+    {:noreply, socket}
```

Comment:
Can we assign the name to the session here instead?

Reply:
It _would_ seem like the natural place, right? But we’re acting on websocket data and only have access to the data sent over the wire and the socket process state. We can’t see the session here, and there are no cookies. 

It made me start thinking of a proposal to LiveView. We do have the session on mount. We could keep it in the socket process state (socket.session), and have a specific api mimicking the regular `Plug.Conn.put_session/3` but working with a liveview socket. Storing something in the session would have the side effect of talking to the client side liveview library to set a new session in the cookie client side. 

Feels doable, but also feels a bit like we’re reimplementing http features we’re missing in web sockets. There can be weird failure states where client and server have different ideas of what’s in the session if that call fails.

Reply:
It _would_ seem like the natural place, right? But we’re acting on websocket data and only have access to the data sent over the wire and the socket process state. We can’t see the session here, and there are no cookies. 

It made me start thinking of a proposal to LiveView. We do have the session on mount. We could keep it in the socket process state (socket.session), and have a specific api mimicking the regular `Plug.Conn.put_session/3` but working with a liveview socket. Storing something in the session would have the side effect of talking to the client side liveview library to set a new session in the cookie client side. 

Feels doable, but also feels a bit like we’re reimplementing http features we’re missing in web sockets. There can be weird failure states where client and server have different ideas of what’s in the session if that call fails.

Reply:
It _would_ seem like the natural place, right? But we’re acting on websocket data and only have access to the data sent over the wire and the socket process state. We can’t see the session here, and there are no cookies. 

It made me start thinking of a proposal to LiveView. We do have the session on mount. We could keep it in the socket process state (socket.session), and have a specific api mimicking the regular `Plug.Conn.put_session/3` but working with a liveview socket. Storing something in the session would have the side effect of talking to the client side liveview library to set a new session in the cookie client side. 

Feels doable, but also feels a bit like we’re reimplementing http features we’re missing in web sockets. There can be weird failure states where client and server have different ideas of what’s in the session if that call fails.

Reply:
Wouldn’t it be nice to just have it as an argument to the handle_event function. Like in mount?

Reply:
Or maybe live view doesn’t send the session cookie ID with every request. Probably not.